src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php

```diff
@@ -291,7 +292,8 @@ protected function triggerEvents(array $events): int
             if (isset($data[EventBehavior::EVENT_CHANGE_ENTITY_ADDITIONAL_VALUES])) {
                 $eventEntityTransfer->setAdditionalValues($data[EventBehavior::EVENT_CHANGE_ENTITY_ADDITIONAL_VALUES]);
             }
-            $eventEntityTransfersByEvent[$data[EventBehavior::EVENT_CHANGE_NAME]][] = $eventEntityTransfer;
+            $keyId = is_int($id) || is_string($id) ? $id : spl_object_id($eventEntityTransfer);
+            $eventEntityTransfersByEvent[$data[EventBehavior::EVENT_CHANGE_NAME]][$keyId] = $eventEntityTransfer;
```

Comment:
A few things with this code that need to be checked.
we have a triggeredRows var lower. Not we can have this number bigger than actual events that we trigger.
If we fix that, please check event records deletion code as it delete those if amount of triggered events in the same as amount of fetched events.


And one more thing that is not related to this code. Could you please check why we have triggerEvents method called and afterward we have in the loop we have a method that trigger events once again and remove them?

Reply:
but what about $triggeredRows var? Now it will lie to us as we trigger less events

Reply:
I got what you mean. we increase triggeredRows for each event, but skip duplicates 
https://github.com/spryker/event-behavior/blob/e87e7ce911155b4cb3054c8c26db62836962f761/src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php#L297

Reply:
Good point. let me adjust code